lib/stm32/common/rtc_common_l1f024.c

```diff
@@ -122,4 +122,189 @@ void rtc_clear_wakeup_flag(void)
 	RTC_ISR &= ~RTC_ISR_WUTF;
 }
 
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the initialization flag
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_set_init_flag(void)
+{
+	RTC_ISR |= RTC_ISR_INIT;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Clears (resets) the initialization flag
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_clear_init_flag(void)
+{
+	RTC_ISR &= ~RTC_ISR_INIT;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Returns if the RTC_ISR init flag RTC_ISR_INITF is set
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+bool rtc_init_flag_is_ready(void)
+{
+	return (RTC_ISR & RTC_ISR_INITF);
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Waits infinitely for initialization flag to be set in RTC_ISR
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_wait_for_init_ready(void)
+{
+	while (!rtc_init_flag_is_ready());
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the bypass shadow bit in RTC_CR
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_enable_bypass_shadow_register(void)
+{
+	RTC_CR |= RTC_CR_BYPSHAD;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Clears the bypass shadow bit in RTC_CR
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_disable_bypass_shadow_register(void)
+{
+	RTC_CR &= ~RTC_CR_BYPSHAD;
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the RTC hour format (24h or 12h)
+
+@details Requires unlocking backup domain write protection (PWR_CR_DBP)
+*/
+void rtc_set_hour_format(enum rtc_fmt fmt)
+{
+	switch (fmt) {
+	case RTC_CR_FMT_24H:
+		RTC_CR &= ~RTC_CR_FMT;
+		break;
+	case RTC_CR_FMT_12H:
+		RTC_CR |= RTC_CR_FMT;
+		break;
+	}
+}
+
+/*---------------------------------------------------------------------------*/
+/** @brief Sets the RTC BCD calendar year value (tens and ones)
+
+@details Requires unlocking the RTC write-protection (RTC_WPR)
+*/
+void rtc_calendar_set_year(uint8_t year_tens, uint8_t year_units)
+{
+	RTC_DR = ((year_tens & RTC_DR_YT_MASK) << RTC_DR_YT_SHIFT) |
+		(RTC_DR & ~(RTC_DR_YT_MASK << RTC_DR_YT_SHIFT));
+	RTC_DR = ((year_units & RTC_DR_YU_MASK) << RTC_DR_YU_SHIFT) |
+		(RTC_DR & ~(RTC_DR_YU_MASK << RTC_DR_YU_SHIFT));
```

Comment:
can this just be
```
RTC_DR &= ~(YT_MASK << YT_SHIFT | YU_MASK << YU_SHIFT);
RTC_DR |= (yt << YT_SHIFT | yu << YU_SHIFT)
```
I don't really see why we need to modify each section separately?

Further, is this a useful user facing api? Do users _want_ to write their own div/mod/log functions to get the different parts of a year out?

Reply:
Ultimately it's difficult to say exactly what users want without taking a poll. I could try to come up with a more effective way of setting the year, and similar functionality.

Reply:
I've added a `_rtc_dec_to_bcd()` static void function as part of rtc_common.c -- I don't expect this to be a user facing function and did not want to make this a define macro. Let me know if you want this changed up